File: 기타/NW/1.js
function solution(k, rates) {
    var answer = 0;
    let cnt = 0;
    let now = k;
    for(let i=0; i<rates.length-1; i++){
        if(cnt === 0 && now >= rates[i]){
            if(rates[i]<rates[i+1]){
                cnt++;
                now -= rates[i];
            }
        }
        else if(cnt === 1){
            if(rates[i]>rates[i+1]){
                cnt--;
                now += rates[i];
            }
        }
    }
    if(cnt === 1)    now += rates[rates.length-1];
    return now;
}

File: 기타/HD/1.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;

vector<vector<int>> dice;
int number[10001];

void dfs(vector<int> visit, int cnt, string now, int want){
    // cout<<cnt<<' '<<now<<' '<<want<<endl;
    // if(cnt==1)  return;
    if(now.length() == want){
        int res = stoi(now);
        number[res]++;
        return;
    }
    for(int i=0; i<dice.size(); i++){
        if(visit[i] == 1)   continue;
        visit[i] = 1;
        for(int j=0; j<dice[i].size(); j++){
            string next = to_string(dice[i][j]);
            dfs(visit, cnt+1, now+next, want);
        }
        visit[i] = 0;
    }
}

int solution(vector<vector<int>> Dice) {
    int answer = 0;
    dice = Dice;
    vector<int> visit;
    for(int i=0; i<=dice.size(); i++){
        visit.push_back(0);
    }
    for(int i=0; i<dice.size(); i++){
        dfs(visit, 0, "", i+1);
    }
    for(int i=1; i<10000; i++){
        if(number[i]==0)    return i;
    }
    return answer;
}